src/main/java/io/fabric8/maven/docker/model/ContainerDetails.java

```diff
@@ -121,6 +126,21 @@ public boolean isRunning() {
         return state.getBoolean(RUNNING);
     }
 
+    @Override
+    public boolean isHealthy() {
+        final JSONObject state = json.getJSONObject(STATE);
+        // always indicate healthy for docker hosts that do not support health checks.
+        return !state.has(HEALTH) || HEALTH_STATUS_HEALTHY.equals(state.getJSONObject(HEALTH).getString(STATUS));
+    }
+
+    @Override
+    public String healthcheckTests() {
+        if (!json.getJSONObject(CONFIG).has(HEALTHCHECK)) {
+            return null;
+        }
+        return json.getJSONObject(CONFIG).getJSONObject(HEALTHCHECK).getJSONArray(TEST).join(", ");
```

Comment:
> So the user configuration can fail for two case: Either no health-check has been provided or the Docker daemon doesn't support health-check. We should report that differently. In the first case its a configuration error (doesn't make sense to wait in this case) in the later it's because a potentially supported configuration is run in an non-compatible environment.

Okay. Since we require a certain server api version for this feature anyway, we could also simply check if the `DockerAccessContext.getMinimalApiVersion()` is configured to >=1.25.0 and advice the user to configure it explicitly otherwise (+ add this requirement to the docs).  What do you think?